File: YFDebugPannel/YFDebugPannel/Utilities/HCValueHelpers.m
#import "HCValueHelpers.h"

NSInteger HCIntValue(id value) {
    if ([value isKindOfClass:[NSNumber class]]) {
        return [value integerValue];
    }
    if ([value isKindOfClass:[NSString class]]) {
        return [(NSString *)value integerValue];
    }
    return 0;
}

BOOL HCBoolValue(id value) {
    if ([value isKindOfClass:[NSNumber class]]) {
        return [value boolValue];
    }
    if ([value isKindOfClass:[NSString class]]) {
        return [(NSString *)value boolValue];
    }
    return NO;
}

File: YFDebugPannel/YFDebugPannel/HCValueHelpers.swift
import Foundation

func intValue(_ value: Any?) -> Int {
    if let value = value as? Int {
        return value
    }
    if let value = value as? NSNumber {
        return value.intValue
    }
    if let value = value as? String, let number = Int(value) {
        return number
    }
    return 0
}

func boolValue(_ value: Any?) -> Bool {
    if let value = value as? Bool {
        return value
    }
    if let value = value as? NSNumber {
        return value.boolValue
    }
    if let value = value as? String {
        return (value as NSString).boolValue
    }
    return false
}
